lighthouse-core/scripts/legacy-javascript/run.js

```diff
@@ -186,65 +72,94 @@ async function createVariant(options) {
-    /** @type {Pick<LH.Artifacts, 'devtoolsLogs'|'URL'|'ScriptElements'>} */
-    const artifacts = {
-      URL: {finalUrl: documentUrl, requestedUrl: documentUrl},
-      devtoolsLogs: {
-        [LegacyJavascript.DEFAULT_PASS]: devtoolsLogs,
-      },
-      ScriptElements: [
-        // @ts-ignore - partial ScriptElement excluding unused DOM properties
-        {requestId: jsRequestId, content: code},
-      ],
-    };
-    // @ts-ignore: partial Artifacts.
-    const legacyJavascriptResults = await LegacyJavascript.audit(artifacts, {
-      computedCache: new Map(),
-    });
+    const code = fs.readFileSync(`${dir}/main.bundle.min.js`, 'utf-8');
+    const map = JSON.parse(fs.readFileSync(`${dir}/main.bundle.min.js.map`, 'utf-8'));
+
+    let legacyJavascriptResults;
+
+    legacyJavascriptResults = await getLegacyJavascriptResults(code, map, {sourceMaps: true});
     fs.writeFileSync(`${dir}/legacy-javascript.json`,
+      // @ts-ignore: Items will exist.
+      JSON.stringify(legacyJavascriptResults.details.items, null, 2));
+
+    legacyJavascriptResults = await getLegacyJavascriptResults(code, map, {sourceMaps: false});
+    fs.writeFileSync(`${dir}/legacy-javascript-nomaps.json`,
+      // @ts-ignore: Items will exist.
       JSON.stringify(legacyJavascriptResults.details.items, null, 2));
   }
 }
 
-function makeSummary() {
+/**
+ * @param {string} code
+ * @param {LH.Artifacts.RawSourceMap} map
+ * @param {{sourceMaps: boolean}} _
+ * @return {Promise<LH.Audit.Product>}
+ */
+function getLegacyJavascriptResults(code, map, {sourceMaps}) {
+  // Instead of running Lighthouse, use LegacyJavascript directly. Requires some setup.
+  // Much faster than running Lighthouse.
+  const documentUrl = 'http://localhost/index.html'; // These URLs don't matter.
+  const scriptUrl = 'https://localhost/main.bundle.min.js';
+  const networkRecords = [
+    {url: documentUrl},
+    {url: scriptUrl},
+  ];
+  const devtoolsLogs = networkRecordsToDevtoolsLog(networkRecords);
+  const jsRequestWillBeSentEvent = devtoolsLogs.find(e =>
+    e.method === 'Network.requestWillBeSent' && e.params.request.url === scriptUrl);
+  if (!jsRequestWillBeSentEvent) throw new Error('jsRequestWillBeSentEvent is undefined');
+  // @ts-ignore - the log event is not narrowed to 'Network.requestWillBeSent' event from find
+  const jsRequestId = jsRequestWillBeSentEvent.params.requestId;
+
+  /** @type {Pick<LH.Artifacts, 'devtoolsLogs'|'URL'|'ScriptElements'|'SourceMaps'>} */
+  const artifacts = {
+    URL: {finalUrl: documentUrl, requestedUrl: documentUrl},
+    devtoolsLogs: {
+      [LegacyJavascript.DEFAULT_PASS]: devtoolsLogs,
+    },
+    ScriptElements: [
+      // @ts-ignore - partial ScriptElement excluding unused DOM properties
+      {src: scriptUrl, requestId: jsRequestId, content: code},
+    ],
+    SourceMaps: [],
+  };
+  if (sourceMaps) artifacts.SourceMaps = [{scriptUrl, map}];
```

Comment:
this is the only meaningful change in here right?

Reply:
yup!